daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/GrammarTerm.scala

```diff
@@ -56,7 +56,7 @@ abstract class Gram(contextArg: SchemaComponent)
   final override lazy val tunable = context.tunable
 
   final override def namespaces = context.namespaces
-  final override def targetNamespace = context.targetNamespace
+  final override def noPrefixNamespace = context.noPrefixNamespace
```

Comment:
So this means at the level of Gram objects, nobody ever asks for noPrefixNamespace. At least in our tests. 

We should either have a test that exercises this, or we should remove the member.  It's likely this is just bypassed, as in the places that could call it are themselves calling noPrefixNamespace on the context object themselves.

Reply:
It looks like not only does nothing ask for `noPrefixNamespace` at the level of `Gram` objects, nothing also asks for `namespaces` and `unqualifiedPathStep` policy, two things also needed for resolving references. It looks like codecov only sees no coverage on noPrefixNamespace for some reason.

I think what that means is we need to move `ResolvesQNames` to a trait down a level so that is only on a DSOM objects but not any Gram objects. Which I think makes sense? DSOM should be resolving everything and then Grams are created based on those results.

I think that means `ResolvesQNames` should be on `SchemaComponent` instead. I'll move it and see what happens.

Reply:
This makes sense to me. DSOM should be the only thing that knows anything about namespaces.  That said, some stuff happens at Gram level that probably should not.

Reply:
I was able to remove ResolvesQNames from the Gram level without issue, it was a relatively small change.